esp/src/src-react/components/FileDetailsGraph.tsx

```diff
@@ -0,0 +1,121 @@
+import { useConst } from "@fluentui/react-hooks";
+import * as Observable from "dojo/store/Observable";
+import { Memory } from "src/Memory";
+import * as Utility from "src/Utility";
+import nlsHPCC from "src/nlsHPCC";
+import { useFile } from "../hooks/File";
+import { HolyGrail } from "../layouts/HolyGrail";
+import { createCopyDownloadSelection, ShortVerticalDivider } from "./Common";
+import { DojoGrid, selector } from "./DojoGrid";
+
+function getStateImageName(row) {
+    if (row.Complete) {
+        return "workunit_completed.png";
+    } else if (row.Running) {
+        return "workunit_running.png";
+    } else if (row.Failed) {
+        return "workunit_failed.png";
+    }
+    return "workunit.png";
+}
+
+const defaultUIState = {
+    hasSelection: false
+};
+
+interface FileDetailsGraphProps {
+    cluster?: string;
+    logicalFile: string;
+}
+
+export const FileDetailsGraph: React.FunctionComponent<FileDetailsGraphProps> = ({
+    cluster,
+    logicalFile
+}) => {
+
+    const [file, , _refresh] = useFile(cluster, logicalFile);
+    const [grid, setGrid] = React.useState<any>(undefined);
+    const [selection, setSelection] = React.useState([]);
+    const [uiState, setUIState] = React.useState({ ...defaultUIState });
+    // const [helpers] = useWorkunitHelpers(wuid);
+
+    //  Command Bar  ---
+    const buttons: ICommandBarItemProps[] = [
+        {
+            key: "refresh", text: nlsHPCC.Refresh, iconProps: { iconName: "Refresh" },
+            onClick: () => refreshTable()
+        },
+        { key: "divider_1", itemType: ContextualMenuItemType.Divider, onRender: () => <ShortVerticalDivider /> },
+        {
+            key: "open", text: nlsHPCC.Open, disabled: !uiState.hasSelection, iconProps: { iconName: "WindowEdit" },
+            onClick: () => {
+                if (selection.length === 1) {
+                    window.location.href = `#/workunits/${file?.Wuid}/graphs/${selection[0].Name}`;
+                } else {
+                    for (let i = 0; i < selection.length; ++i) {
+                        window.open(`#/workunits/${file?.Wuid}/graphs/${selection[i].Name}`, "_blank");
+                    }
+                }
+            }
+        }
+    ];
+
+    const rightButtons: ICommandBarItemProps[] = [
+        ...createCopyDownloadSelection(grid, selection, "graphs.csv")
+    ];
+
+    //  Grid ---
+    const gridStore = useConst(new Observable(new Memory("Name")));
+    const gridQuery = useConst({});
+    const gridColumns = useConst({
+        col1: selector({
+            width: 27,
+            selectorType: "checkbox"
+        }),
+        Name: {
+            label: nlsHPCC.Name, sortable: true,
+            formatter: function (Name, row) {
+                return Utility.getImageHTML(getStateImageName(row)) + `&nbsp;<a href='#/workunits/${file?.Wuid}/graphs/${Name}' onClick='return false;' class='dgrid-row-url'>${Name}</a>`;
```

Comment:
@GordonSmith Hasn't been merged yet it seems, but wondering if the StateIcon component you'd made for https://github.com/hpcc-systems/HPCC-Platform/pull/15042/files could be used here also?

Reply:
Agree - we will be doing a complete change to all icons prior to going gold.

Reply:
Agree - will be doing a full icon refresh before going gold.